eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/DoctrineDatabase.php

```diff
@@ -1443,6 +1443,83 @@ public function repairBrokenUrlAliasesForLocation(int $locationId)
         }
     }
 
+    /**
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function deleteUrlNopAliasesWithoutChildren(): int
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        $queryBuilder = $this->connection->createQueryBuilder();
+
+        // The wrapper select is needed for SQL "Derived Table Merge" issue for deleting
+        $wrapperQueryBuilder = clone $queryBuilder;
+        $selectQueryBuilder = clone $queryBuilder;
+        $expressionBuilder = $queryBuilder->expr();
+
+        $selectQueryBuilder
+            ->select('u_parent.id AS inner_id')
+            ->from($this->table, 'u_parent')
+            ->leftJoin(
+                'u_parent',
+                $this->table,
+                'u',
+                $expressionBuilder->eq('u_parent.id', 'u.parent')
+            )
+            ->where(
+                $expressionBuilder->eq(
+                    'u_parent.action_type',
+                    ':actionType'
+                )
+            )
+            ->groupBy('u_parent.id')
+            ->having(
+                $expressionBuilder->eq($platform->getCountExpression('u.id'),0)
+            );
+
+        $wrapperQueryBuilder
+            ->select('inner_id')
+            ->from(
+                sprintf('(%s) wrapper', $selectQueryBuilder)
```

Comment:
hmm, wrapper is an alias so it should be set as 2nd argument.
Would that work:
```suggestion
                $selectQueryBuilder->getSQL(), 'wrapper'
```
?

Reply:
Almost, below code works:
`sprintf('(%s)', $selectQueryBuilder), 'wrapper'`